overlays/haskell.nix

```diff
@@ -598,17 +620,22 @@ final: prev: {
 
         # Add this to your tests to make all the dependencies of haskell.nix
         # are tested and cached.
-        haskellNixRoots = final.recurseIntoAttrs {
-          Level0 = haskellNixRoots' 0;
-          Level1 = haskellNixRoots' 1;
+        haskellNixRoots = final.haskell-nix.roots (final.haskell-nix.defaultCompilerNixNameWithWarning (default:
+            "Please consider replacing `haskellNixRoots` with `p.roots` on your project "
+          + "or using `haskell-nix.roots \"${default}\"`."));
+        roots = compiler-nix-name: final.recurseIntoAttrs {
+          Level0 = roots' compiler-nix-name 0;
+          Level1 = roots' compiler-nix-name 1;
           # Should be safe to use this now we have materialized everything
           # except the tests
-          Level2 = haskellNixRoots' 2;
+          Level2 = roots' compiler-nix-name 2;
         };
 
-        haskellNixRoots' = ifdLevel:
-          let inherit (final.haskell-nix) defaultCompilerNixName;
-          in final.recurseIntoAttrs ({
+        haskellNixRoots' = final.haskell-nix.roots' (final.haskell-nix.defaultCompilerNixNameWithWarning (default:
+            "Please consider replacing `haskellNixRoots'` with `p.roots'` on your project "
```

Comment:
If we follow the "Rodney" deprecation-strategy then we could just get rid of the warnings and note the change in the changelog.